modules/compute/gke-node-pool/main.tf

```diff
@@ -162,6 +162,13 @@ resource "google_container_node_pool" "node_pool" {
       key                      = var.specific_reservation.key
       values                   = var.specific_reservation.values
     }
+
+    dynamic "host_maintenance_policy" {
```

Comment:
nit: I could not find host_maintenance_policy in https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_node_pool but I do see that it was added https://github.com/hashicorp/terraform-provider-google/pull/15347 , so I might not be looking at the right place.

Reply:
I couldn't find host_maintenance_policy in the docs or Pantheon either. As you mentioned, it is available in the provider repository and has been used in the ai-infra-cluster-provisioning [code](https://github.com/GoogleCloudPlatform/ai-infra-cluster-provisioning/blob/main/a3-mega/terraform/modules/cluster/gke/main.tf#L228). I've applied the changes and verified that these changes are reflected in the gcloud cli response (attached in PR desc.)